forge-ai/src/main/java/forge/ai/AiController.java

```diff
@@ -1304,6 +1304,57 @@ public void declareAttackers(Player attacker, Combat combat) {
         AiAttackController aiAtk = new AiAttackController(attacker); 
         lastAttackAggression = aiAtk.declareAttackers(combat);
 
+        // Check if we can reinforce with Banding creatures
+        if (!combat.getAttackers().isEmpty()) {
+            List<String> bandsWithString = Arrays.asList("Bands with Other Legendary Creatures",
+                    "Bands with Other Creatures named Wolves of the Hunt",
+                    "Bands with Other Dinosaurs");
+            List<Card> bandingCreatures = new CardCollection(aiAtk.notNeededAsBlockers(attacker.getCreaturesInPlay()));
```

Comment:
It's tempting to try and check all available creatures here so you can also catch cases where only a banded attack is acceptable
But this might lead to cheating because the banding creature could have Defender or something?

So probably need to start from the actually selected attackers and respect already made defender decisions
(that is if A attacks PW and B attacks opp don't reroute either into a band as that might just lead to an illegal constraint violation)

Reply:
Good call! I tried to implement a filter where the creatures that are already attacking and/or can't legally attack won't be routed into bands. The addAttacker call already ensures that the target of the banded creature is the same as the target of whatever it's being banded into, so the creatures in the same band shouldn't end up being assigned to different defenders, unless there's a loophole somewhere I don't see?..